### ISSUE TITLE
Dark theme and inline HTML editor white background: how?

### ISSUE THREAD

**Issue by MarvinFS** [roundcube/roundcubemail#8605]
People, I'm sorry, I really have tried to google first, haven't found a working solution.

So, I'm using a dark mode with default elastic theme. Everything is pretty fine, except I'm using HTML emails and whenever I compose new email, I have everything dark except inline html editor's background being pure white, it bothers me much. I've seen some modified themes out here on github, which supposedly fixes that, but they are all outdated now and doesn't work with current roundcube versions, and I have no knowledge to adapt them myself.
What could be done to address that?
And, if possible, all inbound html messages either from INBOX as they all have white background too. 

That what I'm having now: 
[https://imgur.com/a/FKCqTWM](url)

Regards,
MarvinFS

**Comment by alecpl** [comment]
There's no easy solution, because it is not easy to convert HTML content into dark. HTML content can have colors defined everywhere, so you can't just change the background to black and font to white. As for the editor, you can't be sure the recipient's email client supports dark mode and how, so HTML content that works for you might not work for the other side.

**Comment by MarvinFS** [comment]
Aha - I see, yes, that makes sense, although, I believe there could be some clever solutions, still. Like to substitute styles for visual rendering only, leaving raw data and original HTML structures intact. Thank you very much for fast reply, I hope it will be solved some day and for now will have to burn my old eyes with white then! :)

**Comment by modscleo4** [comment]
Outlook just inverts the background color of everything that is not black (and inverts the text color from text inside these components). Also, it has a toggle switch so you can turn the invertion off. It's not perfect for every email, but it's better than a white background on a dark page

**Comment by rpc-scandinavia** [comment]
I just filed a bug report #9940, in there I show how I added dark mode to the mail editor.
This is an answer to the "whenever I compose new email" part of the original post.

**Comment by rpc-scandinavia** [comment]
> EDIT: I made a simple plugin that does this: [https://github.com/modscleo4/roundcube-plugin-dark-html](https://github.com/modscleo4/roundcube-plugin-dark-html/)

Yes, thank you @modscleo4, I use that. Do you want my `nb_NO.inc` and `da_DK.inc` files ?

**Comment by rpc-scandinavia** [comment]
> EDIT: I made a simple plugin that does this: https://github.com/modscleo4/roundcube-plugin-dark-html/

@modscleo4, I used some of your code in my [plugin](https://github.com/rpc-scandinavia/RpcRoundCubeMailPatches/).
